Library/Infrastructure/DefaultNfieldHttpClient.cs

```diff
@@ -0,0 +1,52 @@
+﻿//    This file is part of Nfield.SDK.
+//
+//    Nfield.SDK is free software: you can redistribute it and/or modify
+//    it under the terms of the GNU Lesser General Public License as published by
+//    the Free Software Foundation, either version 3 of the License, or
+//    (at your option) any later version.
+//
+//    Nfield.SDK is distributed in the hope that it will be useful,
+//    but WITHOUT ANY WARRANTY; without even the implied warranty of
+//    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+//    GNU Lesser General Public License for more details.
+//
+//    You should have received a copy of the GNU Lesser General Public License
+//    along with Nfield.SDK.  If not, see <http://www.gnu.org/licenses/>.
+
+using System.Collections.Generic;
+using System.Linq;
+using System.Net.Http;
+using System.Net.Http.Headers;
+using System.Threading.Tasks;
+using Nfield.Extensions;
+
+namespace Nfield.Infrastructure
+{
+    /// <summary>
+    /// A wrapper around <see cref="HttpClient"/> that also adds an authentication header to the response
+    /// if it was received in the request.
+    /// </summary>
+    internal sealed class DefaultNfieldHttpClient : NfieldHttpClientBase
+    {
+        private string _token;
+
+        public DefaultNfieldHttpClient(HttpClient client) : base(client)
+        { }
+
+        public override async Task<HttpResponseMessage> SendAsync(HttpRequestMessage request)
+        {
+            request.Headers.Authorization = new AuthenticationHeaderValue("Basic", _token);
```

Comment:
does this mean the value is null the first time we send it?

Reply:
I suppose it does.. it was already this way (just in another file) and I didn't look too closely at it. Apparently the API ignores this header on the `SignIn` request.